core/src/dispatch/mod.rs

```diff
@@ -851,6 +971,54 @@ impl ComponentLifecycle for NetworkDispatcher {
     }
 }
 
+impl Provide<NetworkStatusPort> for NetworkDispatcher {
+    fn handle(&mut self, event: <NetworkStatusPort as Port>::Request) -> Handled {
+        debug!(
+            self.ctx.log(),
+            "Received NetworkStatusPort Request {:?}", event
+        );
+        match event {
+            NetworkStatusRequest::ConnectedSystems => {
+                let mut response = Vec::new();
+                for (addr, state) in self.connections.iter() {
+                    if let ConnectionState::Connected(_) = state {
+                        response.push(SystemPath::with_socket(Transport::TCP, *addr));
+                    }
+                }
+                self.network_status_port
+                    .trigger(NetworkStatus::ConnectedSystems(response));
+            }
+            NetworkStatusRequest::DisconnectedSystems => {
```

Comment:
Does it make sense to just group these types together without indicating what their state is? Maybe we should just reply with the connection state and only have one event for all states?